### ISSUE TITLE
Server Settings not working when accessing externally

### ISSUE THREAD

**Issue by JohnDoe-dev-spec** [Wizarrrr/wizarr#22]
Hi,

I noticed that when i attempt to modify the server settings and scan for libraries, it always comes back with error message that the PLEX URL and token may be incorrect.

However, when accessing Wizarr Server settings via my local IP address, the same Plex URL and token are used but this time it works and i can see all the libraries as well as verify and confirm.

Initially i thought it may be an issue that Authentik may be causing, so i tried disabling this so that the site is hosted directly, but i'm afraid its the same resulting error.

Not sure if others are experiencing this or not.

Thanks

**Comment by suckerfish** [comment]
I'm experiencing the opposite. My publicly accessible Plex URL/address works fine. Pointing Wizarr to the internal address doesn't work. Maybe a higher level of debug added to the logs would be helpful in this case to find out what exactly is going on.

**Comment by mtthidoteu** [comment]
> Hi,
> 
> I noticed that when i attempt to modify the server settings and scan for libraries, it always comes back with error message that the PLEX URL and token may be incorrect.
> 
> However, when accessing Wizarr Server settings via my local IP address, the same Plex URL and token are used but this time it works and i can see all the libraries as well as verify and confirm.
> 
> Initially i thought it may be an issue that Authentik may be causing, so i tried disabling this so that the site is hosted directly, but i'm afraid its the same resulting error.
> 
> Not sure if others are experiencing this or not.
> 
> Thanks

Would you mind pasting your browser console log when trying to Scan the Plex Libraries? (Inspectent element --> Console)

**Comment by JohnDoe-dev-spec** [comment]
> > Hi,
> > I noticed that when i attempt to modify the server settings and scan for libraries, it always comes back with error message that the PLEX URL and token may be incorrect.
> > However, when accessing Wizarr Server settings via my local IP address, the same Plex URL and token are used but this time it works and i can see all the libraries as well as verify and confirm.
> > Initially i thought it may be an issue that Authentik may be causing, so i tried disabling this so that the site is hosted directly, but i'm afraid its the same resulting error.
> > Not sure if others are experiencing this or not.
> > Thanks
> 
> Would you mind pasting your browser console log when trying to Scan the Plex Libraries? (Inspectent element --> Console)

(index):62 cdn.tailwindcss.com should not be used in production. To use Tailwind CSS in production, install it as a PostCSS plugin or use the Tailwind CLI: https://tailwindcss.com/docs/installation
(anonymous) @ (index):62
(index):97          POST https://wizarr.mydomain.com/scan?plex_url=http://PLEXIPAddress:42401/&plex_token=MYTOKEN 403
scanLibraries @ (index):97
onclick @ (index):65

**Comment by mtthidoteu** [comment]
> > > Hi,
> > > I noticed that when i attempt to modify the server settings and scan for libraries, it always comes back with error message that the PLEX URL and token may be incorrect.
> > > However, when accessing Wizarr Server settings via my local IP address, the same Plex URL and token are used but this time it works and i can see all the libraries as well as verify and confirm.
> > > Initially i thought it may be an issue that Authentik may be causing, so i tried disabling this so that the site is hosted directly, but i'm afraid its the same resulting error.
> > > Not sure if others are experiencing this or not.
> > > Thanks
> > 
> > 
> > Would you mind pasting your browser console log when trying to Scan the Plex Libraries? (Inspectent element --> Console)
> 
> (index):62 cdn.tailwindcss.com should not be used in production. To use Tailwind CSS in production, install it as a PostCSS plugin or use the Tailwind CLI: https://tailwindcss.com/docs/installation (anonymous) @ (index):62 (index):97 POST https://wizarr.mydomain.com/scan?plex_url=http://PLEXIPAddress:42401/&plex_token=MYTOKEN 403 scanLibraries @ (index):97 onclick @ (index):65

Okay! So it is giving you a 403 error, meaning the POST request made to the `/scan` endpoint was denied. Try going to wizarr/scan. If it is successful, you will get a 'Method not allowed', if it is blocked by something, you should get a 'Unauthorised` error message. The latter would imply that your reverse proxy (or something) is blocking your browser from accessing /scan.

**Comment by JohnDoe-dev-spec** [comment]
> > > > Hi,
> > > > I noticed that when i attempt to modify the server settings and scan for libraries, it always comes back with error message that the PLEX URL and token may be incorrect.
> > > > However, when accessing Wizarr Server settings via my local IP address, the same Plex URL and token are used but this time it works and i can see all the libraries as well as verify and confirm.
> > > > Initially i thought it may be an issue that Authentik may be causing, so i tried disabling this so that the site is hosted directly, but i'm afraid its the same resulting error.
> > > > Not sure if others are experiencing this or not.
> > > > Thanks
> > > 
> > > 
> > > Would you mind pasting your browser console log when trying to Scan the Plex Libraries? (Inspectent element --> Console)
> > 
> > 
> > (index):62 cdn.tailwindcss.com should not be used in production. To use Tailwind CSS in production, install it as a PostCSS plugin or use the Tailwind CLI: https://tailwindcss.com/docs/installation (anonymous) @ (index):62 (index):97 POST https://wizarr.mydomain.com/scan?plex_url=http://PLEXIPAddress:42401/&plex_token=MYTOKEN 403 scanLibraries @ (index):97 onclick @ (index):65
> 
> Okay! So it is giving you a 403 error, meaning the POST request made to the `/scan` endpoint was denied. Try going to wizarrurl.com/scan. If it is successful, you will get a 'Method not allowed' (because you are not sending a proper request), if it is blocked by something, you should get a 'Unauthorised` error message. The latter would imply that your reverse proxy (or something) is blocking your browser from accessing /scan.
> 
> If you are using Authentik, I would imagine that if you are in the settings, you would be logged in anyway, so it is fine(and better) to leave /scan behind the Auth.

I get the following when going to wizarr/scan from either my external facing url, or local IP address.

405 Method Not Allowed
The method is not allowed for the requested URL.

**Comment by mtthidoteu** [comment]
> > > > > Hi,
> > > > > I noticed that when i attempt to modify the server settings and scan for libraries, it always comes back with error message that the PLEX URL and token may be incorrect.
> > > > > However, when accessing Wizarr Server settings via my local IP address, the same Plex URL and token are used but this time it works and i can see all the libraries as well as verify and confirm.
> > > > > Initially i thought it may be an issue that Authentik may be causing, so i tried disabling this so that the site is hosted directly, but i'm afraid its the same resulting error.
> > > > > Not sure if others are experiencing this or not.
> > > > > Thanks
> > > > 
> > > > 
> > > > Would you mind pasting your browser console log when trying to Scan the Plex Libraries? (Inspectent element --> Console)
> > > 
> > > 
> > > (index):62 cdn.tailwindcss.com should not be used in production. To use Tailwind CSS in production, install it as a PostCSS plugin or use the Tailwind CLI: https://tailwindcss.com/docs/installation (anonymous) @ (index):62 (index):97 POST https://wizarr.mydomain.com/scan?plex_url=http://PLEXIPAddress:42401/&plex_token=MYTOKEN 403 scanLibraries @ (index):97 onclick @ (index):65
> > 
> > 
> > Okay! So it is giving you a 403 error, meaning the POST request made to the `/scan` endpoint was denied. Try going to wizarrurl.com/scan. If it is successful, you will get a 'Method not allowed' (because you are not sending a proper request), if it is blocked by something, you should get a 'Unauthorised` error message. The latter would imply that your reverse proxy (or something) is blocking your browser from accessing /scan.
> > If you are using Authentik, I would imagine that if you are in the settings, you would be logged in anyway, so it is fine(and better) to leave /scan behind the Auth.
> 
> I get the following when going to wizarr/scan from either my external facing url, or local IP address.
> 
> 405 Method Not Allowed The method is not allowed for the requested URL.


That is a good sign! Okay, then it is not an issue with your reverse proxy. The 403 must be coming from WIzarr talking to the plex server and getting an error because the Token is not valid (hence the 403 'Unauthorised'). Have you definitely followed https://support.plex.tv/articles/204059436-finding-an-authentication-token-x-plex-token/ ?

**Comment by JohnDoe-dev-spec** [comment]
> That is a good sign! Okay, then it is not an issue with your reverse proxy. The 403 must be coming from WIzarr talking to the plex server and getting an error because the Token is not valid (hence the 403 'Unauthorised'). Have you definitely followed https://support.plex.tv/articles/204059436-finding-an-authentication-token-x-plex-token/ ?

Yeah, definitely followed it correctly, as if I were to use the exact same token in http://192.168.10.50:5690 it is able to scan for the libraries no problem.

I modified my /etc/hosts file on my Windows machine and added the following entry

192.168.10.50 wizarr-test.local

Then navigated to http://wizarr-test.local:5690 and the scanning for library worked ok this way too.

It does seem to be that when using any URLs via my reverse proxy it stops working properly, 


Before I decided to post this message, I thought I'd take another look at the reverse proxy and managed to now get it working!!. 

It turns out enabling the 'Block Common Exploits' is what caused all the issues.

I wonder if there is something you're able to do that'll let it work when that option is enabled?
